pkg/serverless/trace/inferredspan/inferred_span.go

```diff
@@ -72,3 +83,66 @@ func FilterFunctionTags(input map[string]string) map[string]string {
 
 	return output
 }
+
+func CreateInferredSpan(event string, inferredSpan InferredSpan) {
+	// Parse the event into the EventKey struct
+	eventSource, attributes := ParseEventSource(event)
+	switch eventSource {
+	case "apigateway":
+		CreateInferredSpanFromAPIGatewayEvent(eventSource, attributes, inferredSpan)
+	case "http-api":
+		CreateInferredSpanFromAPIGatewayHTTPEvent(eventSource, attributes, inferredSpan)
+	case "websocket":
+		CreateInferredSpanFromAPIGatewayWebsocketEvent(eventSource, attributes, inferredSpan)
+	}
+}
+
+func CompleteInferredSpan(
+	processTrace func(p *api.Payload),
+	endTime time.Time,
+	isError bool,
+	requestId string,
+	inferredSpan InferredSpan) {
+
+	if inferredSpan.IsAsync {
+		inferredSpan.Span.Duration = inferredSpan.Span.Start
```

Comment:
Not sure I understand this -- if the span is async, the duration is equal to the start time? 🤔